daemon/graphdriver/aufs/aufs.go

```diff
@@ -350,6 +350,10 @@ func (a *Driver) Put(id string) error {
 // Diff produces an archive of the changes between the specified
 // layer and its parent layer which may be "".
 func (a *Driver) Diff(id, parent string) (archive.Archive, error) {
+	// Make sure the hardlinks are copied up before taking the diff
+	if err := exec.Command("auplink", path.Join(a.rootPath(), "mnt", id), "flush").Run(); err != nil {
```

Comment:
do you think this might be better in `func a.unmount` since this flush is needed before every unmount?